### PR TITLE
[test] add test for api

### PR DESCRIPTION
**Pull request by huof6829**
add test for server_V2
ignore api.go for nouse

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -10,13 +10,14 @@ coverage:
...
-  - "action/protocol/account/accountpb"
-  - "action/protocol/poll/*.abi.go"
-  - "blockindex/indexpb"
-  - "consensus/scheme/rolldpos/endorsementpb"
-  - "db/trie/triepb"
-  - "e2etest/*"
-  - "ioctl/**/*"
-  - "test/mock/**/*"
-  - "tools/**/*"
+  - 'action/protocol/rewarding/rewardingpb'
```
Comment (Liuhaai) [inline .codecov.yml:13]: this change is unneeded

**Review by Liuhaai — APPROVED** [review]

File: api/serverV2_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	require := require.New(t)
+	cfg := newConfig(t)
+	config.SetEVMNetworkID(1)
+	svr, bfIndexFile, _ := createServerV2(cfg, false)
+	defer func() {
+		testutil.CleanupPath(t, bfIndexFile)
+	}()
+
+	err := svr.Start()
+	require.NoError(err)
```
Comment (Liuhaai) [inline api/serverV2_test.go:23]: should start() end stop() without no error

**Review by Liuhaai — CHANGES_REQUESTED** [review]

File: .codecov.yml (hunk 1 above)
Comment (huof6829) [reply .codecov.yml]: yes

File: api/serverV2_test.go (hunk 2 above)
Comment (huof6829) [reply api/serverV2_test.go:24]: yes

File: api/serverV2_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	time.Sleep(3 * time.Second)
```
Comment (dustinxie) [inline api/serverV2_test.go:26]: does a short time like 300ms work?

File: api/serverV2_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+	svr.core.chainListener = nil
+	err := svr.Start()
+	require.Error(err)
+	require.Contains(err.Error(), "failed to add chainListener")
+
+	svr.tracer = &tracesdk.TracerProvider{}
+	err = svr.Stop()
+	require.Error(err)
+	require.Contains(err.Error(), "failed to shutdown api tracer")
```
Comment (dustinxie) [inline api/serverV2_test.go:48]: can combine these 2 tests into `TestServerV2Start` above? the `createServerV2` part is identical

**Review by dustinxie — APPROVED** [review]
approved with 1 nit

File: api/serverV2_test.go (hunk 3 above)
Comment (Liuhaai) [reply api/serverV2_test.go:26]: same flaky problem as https://github.com/iotexproject/iotex-core/pull/2898
pls fix this problem before merge

Comment (huof6829) [reply api/serverV2_test.go:26]: ok

File: api/serverV2_test.go (hunk 4 above)
Comment (huof6829) [reply api/serverV2_test.go:48]: ok

File: api/serverV2_test.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	svr, bfIndexFile, _ := createServerV2(cfg, false)
+	defer func() {
+		testutil.CleanupPath(t, bfIndexFile)
+	}()
+
+	err := svr.Start()
+	require.NoError(err)
+
+	err = testutil.WaitUntil(100*time.Millisecond, 3*time.Second, func() (bool, error) {
+		err = svr.Stop()
```
Comment (Liuhaai) [inline api/serverV2_test.go:27]: don't use `svr.Stop() == nil` to assure the start of server

Comment (huof6829) [reply api/serverV2_test.go:27]: this test  checks whether running start and stop  can make errors through default port 14014 and 15014.   but the /readiness is through another port,  and it checks wether the machine is ready or running health before iotex-server starts, then generate metrics through default port 8080.   so  i think  add this is meaningless. and you ?

Comment (Liuhaai) [reply api/serverV2_test.go:27]: oh, two `server` got mixed up

**Pull request merged**